src/app/api/repositories/[owner]/[repo]/preferences/route.ts

```
@@ -1,61 +1,23 @@
-  { params }: { params: { owner: string; repo: string } }
+  context: { params: { owner: string; repo: string } }
 ) {
   try {
-    const session = await getServerSession(authOptions);
-    if (!session?.user?.email) {
-      return new Response("Unauthorized", { status: 401 });
-    }
-
+    const { session } = await validateSession();
+    const { owner, repo } = await context.params;
+    const [userId, repoName] = [session.user?.id!, `${owner}/${repo}`];
```

ESLint

[error] Optional chain expressions can return undefined by design - using a non-null assertion is unsafe and wrong.

(@typescript-eslint/no-non-null-asserted-optional-chain)